src/App.tsx

```diff
@@ -8,7 +8,65 @@ import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
 
+import { Todo } from './types/Todo';
+import { FilterBy } from './types/FilterBy';
+import { TodoStatus } from './types/TodoStatus';
+
+import { getTodos } from './api';
+
+const getFilteredTodos = (todos: Todo[], query: string, filterBy: FilterBy) => {
+  let newTodos = [...todos];
+
+  if (query) {
+    const newQuery = query.toLowerCase().trim();
+
+    newTodos = newTodos.filter(todo => todo.title.toLowerCase().includes(newQuery));
+  }
+
+  switch (filterBy) {
+    case TodoStatus.Active:
+      return newTodos.filter(todo => !todo.completed);
+
+    case TodoStatus.Completed:
+      return newTodos.filter(todo => todo.completed);
+
+    default:
+      break;
+  }
+
+  return newTodos;
+};
+
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [query, setQuery] = useState('');
+  const [filterBy, setFilterBy] = useState<FilterBy>('all');
+  const [isLoading, setIsLoading] = useState(true);
+  const [error, setError] = useState(false);
+
+  const fetchTodos = async () => {
+    try {
+      const arrayTodos = await getTodos();
+
+      setTodos(arrayTodos);
+      setIsLoading(false);
+      setError(false);
+    } catch {
+      setError(true);
+    }
+  };
+
+  useEffect(() => {
+    fetchTodos();
+  }, []);
+
+  const visibleTodos = getFilteredTodos(todos, query, filterBy);
+
+  const closeTodo = () => {
```

Comment:
Do you close todo? Or maybe you close a modal window?
Use more descriptive names